probe-rs/src/architecture/xtensa/mod.rs

```diff
@@ -101,6 +105,53 @@ impl<'probe> Xtensa<'probe> {
 
         Ok(())
     }
+
+    /// Check if the current breakpoint is a semihosting call
+    // OpenOCD implementation: https://github.com/espressif/openocd-esp32/blob/93dd01511fd13d4a9fb322cd9b600c337becef9e/src/target/espressif/esp_xtensa_semihosting.c#L42-L103
+    fn check_for_semihosting(
+        old_reason: HaltReason,
+        core: &mut dyn CoreInterface,
+    ) -> Result<HaltReason, Error> {
+        let mut reason = old_reason;
+        let pc: u32 = core.read_core_reg(core.program_counter().id)?.try_into()?;
+
+        let mut actual_instructions = [0u8; 3];
+        core.read_8((pc) as u64, &mut actual_instructions)?;
+
+        tracing::debug!(
+            "Semihosting check pc={pc:#x} instructions={0:#08x} {1:#08x} {2:#08x}",
+            actual_instructions[0],
+            actual_instructions[1],
+            actual_instructions[2],
+        );
+
+        let mut expected_instruction = vec![];
+        Instruction::Break(1, 14).encode_into_vec(&mut expected_instruction);
+        let expected_instruction: [u8; 3] = [
+            expected_instruction[0],
```

Comment:
and this we need?

Reply:
Yes, Xtensa instructions are 3 bytes (sometimes 2), we need to chop off the last byte and not use it in the comparison. With that said, we can probably use `to_bytes` and do some slicing.